modules/extensions/src/data-filter/shader-module.ts

```diff
@@ -48,45 +58,48 @@ float dataFilter_reduceValue(vec3 value) {
 float dataFilter_reduceValue(vec4 value) {
   return min(min(value.x, value.y), min(value.z, value.w));
 }
-void dataFilter_setValue(DATAFILTER_TYPE valueFromMin, DATAFILTER_TYPE valueFromMax, DATACATEGORY_TYPE category) {
-  if (filter_enabled) {
+
+#ifdef DATAFILTER_ATTRIB
+  void dataFilter_setValue(DATAFILTER_TYPE valueFromMin, DATAFILTER_TYPE valueFromMax) {
     if (filter_useSoftMargin) {
       dataFilter_value = dataFilter_reduceValue(
         smoothstep(filter_min, filter_softMin, valueFromMin) *
-        (1.0 - smoothstep(filter_softMax, filter_max, valueFromMax))
+          (1.0 - smoothstep(filter_softMax, filter_max, valueFromMax))
       );
     } else {
       dataFilter_value = dataFilter_reduceValue(
         step(filter_min, valueFromMin) * step(valueFromMax, filter_max)
       );
     }
+  }
+#endif
 
+#ifdef DATACATEGORY_ATTRIB
```

Comment:
While there is some possible performance gains from removing the entire function, I wonder if we could just have commented out the parts of it that are dependent on conditional attributes in order to reduce the combinatorial explosion of various code paths - each of this has the potential to be invisible to the compiler in our normal test cases and therefore could be prone to "code rot".

Reply:
I considered that, but then it is messy because there are many combinations of function parameters that may or may not be needed depending on the enabled filter